tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Azure.AI.OpenAI;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Services;
+using OpenAI.Chat;
+
+namespace Azure.Sdk.Tools.Cli.Tools
+{
+    // Will add after we decide which tools are exported via the MCP.
+    //[McpServerToolType, Description("Generates a README file, using service documentation")]
+    public partial class ReadMeGeneratorTool : MCPTool
+    {
+        private readonly ILogger<ReadMeGeneratorTool> logger;
+        private readonly IOutputService output;
+        private readonly AzureOpenAIClient openAiClient;
+        private Option<string> packagePathOption;
+        private Option<string> outputPathOption;
+        private Option<string> templatePathOption;
+        private Option<string> serviceDocumentationOption;
+        private Option<string> modelOption;
+
+        public ReadMeGeneratorTool(ILogger<ReadMeGeneratorTool> logger, IOutputService output, AzureOpenAIClient openAiClient)
+        {
+            ArgumentNullException.ThrowIfNull(logger, nameof(logger));
+            ArgumentNullException.ThrowIfNull(output, nameof(output));
```

Comment:
I don't think these are necessary, you can trust the dependency injector, if they're null we'll have bigger problems. Or is this a check for test instantiation?

Reply:
That's a good point, I probably don't need those - I believe they got added "automatically".

Reply:
I removed them, and nothing exploded so I think we're good.